### PR TITLE
Update openMS, renaming (sub)processes and inclusion of cometAdapter output

### PR DESCRIPTION
**Pull request by marissaDubbelaar**
### `Added`

- Additional output from `CometAdapter` (generated with the parameter `--pin_out`)
- Folder structure within the `intermediate_results` folder to retrace the outcome files easier
- `OPENMS_FALSEDISCOVERYRATE` and `OPENMS_IDFILTER_FOR_ALIGNMENT` are now included in the first quantification step
- Altered the outcome content with the inclusion of the different folder structure
- Updated the mhcquant_web.png in the `assets` folder

### `Fixed`

- Resolved issues with `SAMPLESHEET_CHECK`
- Fix for the `peakpickerhires`, mzml files generated from input raw files are now seen as input for this step as well
- `PRE_QUANTIFICATION` is renamed to `MAP_ALIGNMENT` to indicate that the alignment (and the complementing processes) of the different maps happens here
- `POST_QUANTIFICATION` is renamed to `PROCESS_FEATURE` since the feature identification and processing is done here
- Outcome of `OPENMS_FEATUREFINDERIDENTIFICATION` got lost during one of the previous updates, this is reintroduced
- `OPENMS_TEXTEXPORTER_UNQUANTIFIED` and `OPENMS_TEXTEXPORTER_QUANTIFIED` return only significant hits again

### `Dependencies`

- Updated the multiQC module

| Dependency | Old version | New version |
| ---------- | ----------- | ----------- |
| `MultiQC`  | 1.11        | 1.12        |
| `OpenMS`   | 2.6         | 2.8         |
| `OpenMS thirdparty`   | 2.6         | 2.8      |

### `Deprecated`

- `OPENMS_TEXTEXPORTER_PSMS` was removed due to the outcome of the comet adapter step

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -129,46 +132,48 @@ process {
                 (params.fdr_level != 'psm-level-fdrs') ? "-" + params.fdr_level : ""
```
Comment (jonasscheid) [inline conf/modules.config:132]: Not related to the PR, but in general: Why do we use 5% FDR in Percolator but we are filtering with 1% FDR in OPENMS_IDFILTER_FOR_ALIGNMENT and OPENMS_IDFILTER_Q_VALUE

File: conf/modules.config [hunk 2]
```diff
@@ -192,14 +197,18 @@ process {
...
                 ext.prefix  = { "${meta.sample}_${meta.condition}_all_ids_merged" }
                 publishDir  = [
-                    path: { "${params.outdir}/Intermediate_Results" },
+                    path: { "${params.outdir}/intermediate_results/refined_fdr" },
                     mode: params.publish_dir_mode,
                     pattern: '*.mzTab'
                 ]
             }
 
         withName: MHCFLURRY_PREDICTPSMS {
```
Comment (jonasscheid) [inline conf/modules.config:206]: Name is a little bit ambigous imo. We don't predict PSMs, we predict the peptides resulting from the peptide spectrum match. MHCFLURRY_PREDICT_PEPTIDES makes more sense to me

File: docs/output.md [hunk 3]
```diff
@@ -60,14 +60,25 @@ PEPTIDE contains information about peptide hits that were identified and corresp
...
+  - `features`
+    - `*.mztab` : mztab file generated by the OpenMS MzTabExporter command which is present in the `PROCESS_FEATURE` step
+    - `*.idXML` : Outcome of `PSMFEATUREEXTRACTOR`, containing the computations of extra features for each input PSM
+    - `*.featureXML` : These files file is generated by the OpenMS `FeatureFinderIdentification` command
+  - `percolator`
+    - `*all_ids_merged_psm_perc.idXML` : idXML files are generated with `OPENMS_PERCOLATORADAPTER`
+  - `refined_fdr` (Only if `--refine_fdr_on_predicted_subset` is specified)
+    - `*merged_psm_perc_filtered.mzTab` : This file export filtered percolator results (by q-value) as mztab
+    - `*_all_ids_merged.mzTab` : Exportas all of the psm results as mztab
+    - `*perc_subset.idXML` : This file is the outcome of a second OpenMS `PercolatorAdapter` run
```
Comment (jonasscheid) [inline docs/output.md:79]: We run percolator multiple times :O?
Why?

File: lib/WorkflowMain.groovy [hunk 4]
```diff
@@ -79,17 +79,4 @@ class WorkflowMain {
             System.exit(1)
         }
     }
-
```
Comment (jonasscheid) [inline lib/WorkflowMain.groovy:82]: Why is this removed?

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,16 +4,35 @@
 
 ### `Added`
 
+- Additional output from `CometAdapter` (generated with the parameter `--pin_out`)
```
Comment (jonasscheid) [inline CHANGELOG.md:7]: Attached Issues are missing here?

Comment (jonasscheid) [inline CHANGELOG.md:7]:
```suggestion
- Additional output from `CometAdapter` (generated with the parameter `-pin_out`)
```

**Review by jonasscheid — APPROVED** [review]
LGTM! Maybe we can go over some of my questions and resolve them before the merge :)

File: CHANGELOG.md (hunk 5 above)
Comment (marissaDubbelaar) [reply CHANGELOG.md:7]: Based on an internal evaluation we discussed that this might be of interest to add to the comet outcome. Here all of the peptide hits are present and can be used to obtain the PSMs from the identified peptides.
There is indeed no issue, we can still generate that though (same as for all of the other points).

File: docs/output.md (hunk 3 above)
Comment (marissaDubbelaar) [reply docs/output.md:79]: Percolator can be run two times when the parameters `--refine_fdr_on_predicted_subset` and `predict_class_1 ` are provided. Here, Percolator is used on the subset of predicted binders which results in an increased identification rate, hence the second iteration.

File: conf/modules.config (hunk 2 above)
Comment (marissaDubbelaar) [reply conf/modules.config:206]: I do not understand what you want to mention here regarding the naming @jonasscheid. About the order of the process I would be open for suggestions

File: lib/WorkflowMain.groovy (hunk 4 above)
Comment (marissaDubbelaar) [reply lib/WorkflowMain.groovy:82]: Because this test is giving warnings the whole time and checks according to the NGS genome.

File: subworkflows/local/map_alignment.nf [hunk 6]
```diff
@@ -2,22 +2,31 @@
  * Perform the quantification of the samples when the parameter --skip_quantification is not provided
```
[the remark this answers is not in the document]
Comment (marissaDubbelaar) [reply subworkflows/local/map_alignment.nf:2]: Overall you want to have the quantification step since this is the aspect that aligns the spectra among the different runs. Making the outcome of the data more accurate as well. However, for some use cases you do not want to have the quantification (e.g. when the samples are not that well or if the output does not require quantification). Still, our quantification is the default, since this results in the expected outcome.

**Comment by marissaDubbelaar** [comment]
@jonasscheid let me know if your questions are resolved or if we need to look into it in more detail

File: conf/modules.config (hunk 2 above)
Comment (jonasscheid) [reply conf/modules.config:206]: No I think it is fine like that regarding the naming. Now I understood it better!
